test/common/index.js

```diff
@@ -812,7 +812,49 @@ function restoreWritable(name) {
   delete process[name].writeTimes;
 }
 
+function onResolvedOrRejected(promise, callback) {
```

Comment:
Can you call this `finally` as per https://github.com/tc39/proposal-promise-finally